### DISCUSSION TITLE
Autoregister for reflection

### DISCUSSION THREAD

**Discussion by bulldog98** [quarkusio/quarkus#33688] [Q&A]
In the case of a http client or an Emitter connected to a json smallrye messaging channel it would make sense to register the types for reflection, is there a reason why Quarkus does not do so already?

**Comment by bulldog98** [comment]
\cc @maxandersen this one did not get the smallrye tag, does quarkus-bot only check the title?

**Reply by gsmet** [reply]
Could you please give an example of what you are doing and what is not registered for reflection? That would be helpful.

**Reply by bulldog98** [reply]
If you have an class like this
```java
@Singleton
public class StartUpInitialization {
    private final Emitter<SomeType> requestDevices;

    public StartUpInitialization(@Channel("some-channel") final Emitter<SomeType> requestDevices) {
        this.requestDevices = requestDevices;
    }
    
    // do something with the emitter
}
```
then it is my understanding that `SomeType` is not automatically registered for reflection
